java/arcs/android/common/SqlUtils.kt

```diff
@@ -26,3 +27,36 @@ inline fun <T : Any?> SQLiteDatabase.transaction(block: SQLiteDatabase.() -> T):
         endTransaction()
     }
 }
+
+/**
+ * Helper function for iterating over each row in the results of a query.
+ *
+ * Usage:
+ *
+ * ```kotlin
+ * database.rawQuery("SELECT name, age FROM users", emptyArray()).forEach {
+ *     ageMap[it.getString(0)] = it.getLong(1)
+ * }
+ * ```
+ */
+inline fun Cursor.forEach(block: (Cursor) -> Unit) = use {
```

Comment:
It might be good to note in the KDoc that this will close the Cursor after execution.

Reply:
Done